<input type="number">: a11y and when not to use (#14911)
mdn/content: files/en-us/web/html/element/input/number/index.md

## Using number inputs

> The number input type should only be used for incremental numeric numbers, especially when spinbutton incrementing and decrementing are helpful to user experience. The number input is not appropriate for values that happen to only consist of numbers but aren't strictly speaking a number, such as postal codes in many countries or credit card numbers. For non-numeric numerical inputs, consider using a different input type, such as [`<input type="tel">`](/en-US/docs/Web/HTML/Element/input/tel) or other {{HTMLElement('input')}} with the {{HTMLAttrXref('inputmode')}} attribute:

Reviewer: ```suggestion
The `number` input type should only be used for incremental numbers, especially when spinbutton incrementing and decrementing are helpful to user experience. The `number` input type is not appropriate for values that happen to only consist of numbers but aren't strictly speaking a number, such as postal codes in many countries or credit card numbers. For non-numeric inputs, consider using a different input type, such as [`<input type="tel">`](/en-US/docs/Web/HTML/Element/input/tel) or other {{HTMLElement('input')}} type with the {{HTMLAttrXref('inputmode')}} attribute:
```

>
> If you didn't do this, you'd have to fill in both feet/inches **and** meters to submit the form!

> ## Accessibility

Reviewer: Should this section be moved up before the "Examples" section, so that there is consistency with the following order on other pages: "Examples" - "Specifications" - "Browser compatibility"?

Author: I put it after because it is associated to input type=number, but not part of the explanation. if it need to be above for consistency, it can be moved, but it makes more sense to me to keep it after examples and before specifications, which is where we put all the a11y in 2018, but not sure if things changed site wide since then

Reviewer: Ah ok. I was not aware that that's where we add the a11y content. Guess we'll let it be there then. Thanks!

> The implicit [role](/en-US/docs/Web/Accessibility/ARIA/Roles) for `<input type="number">` is [`spinbutton`](/en-US/docs/Web/Accessibility/ARIA/Roles/spinbutton). If the spinbutton is not an important feature for your form control, consider *not* using `type="number"`, opting for [`inputmode="numeric"`](/en-US/docs/Web/HTML/Global_attributes/inputmode) along with a [`pattern`](/en-US/docs/Web/HTML/Attributes/pattern) attribute limiting the characters to numbers and associated characters instead. With `<input type="number">` there is a risk of users accidentally incrementing a number when they’re trying to do something else and, if the user tries to enter something that’s not a number, there’s no explicit feedback about what they’re doing wrong.

Reviewer: ```suggestion
The implicit [role](/en-US/docs/Web/Accessibility/ARIA/Roles) for the `<input type="number">` element is [`spinbutton`](/en-US/docs/Web/Accessibility/ARIA/Roles/spinbutton). If spinbutton is not an important feature for your form control, consider *not* using `type="number"`. Instead, use [`inputmode="numeric"`](/en-US/docs/Web/HTML/Global_attributes/inputmode) along with a [`pattern`](/en-US/docs/Web/HTML/Attributes/pattern) attribute that limits the characters to numbers and associated characters. With `<input type="number">`, there is a risk of users accidentally incrementing a number when they’re trying to do something else. Additionally, if users try to enter something that’s not a number, there’s no explicit feedback about what they’re doing wrong.
```

> Also consider using the [`autocomplete`](/en-US/docs/Web/HTML/Attributes/autocomplete) attribute to help users complete forms more quickly and with fewer chances of errors. For example, to enable autofill on a zip code field, set `autocomplete="postal-code"`

Reviewer: ```suggestion
Also consider using the [`autocomplete`](/en-US/docs/Web/HTML/Attributes/autocomplete) attribute to help users complete forms more quickly and with fewer chances of errors. For example, to enable autofill on a zip code field, set `autocomplete="postal-code"`.
```
